### PR TITLE
feat(hogql): remove pydantic from AST

### PR DESCRIPTION
**Pull request by mariusandra**
## Problem

Parsing and validating the HogQL AST takes a lot of time due to some inefficient Python code. Namely we're creating and destroying a lot of Pydantic classes. Pydantic includes a lot of run-time validation that we don't need.

## Changes

Swaps from Pydantic to Python Dataclasses.

Before:

After:

## How did you test this code?

Ran all the tests. Nothing seemed broken in the interface. There _might_ be really edge case queries that worked before, but fail now. Let's keep eyes open for them.

### CODE REVIEW

File: posthog/hogql/visitor.py [hunk 1]
```diff
@@ -377,7 +377,7 @@ def visit_field(self, node: ast.Field):
             start=None if self.clear_locations else node.start,
             end=None if self.clear_locations else node.end,
             type=None if self.clear_types else node.type,
-            chain=node.chain,
+            chain=node.chain.copy(),
```
Comment (mariusandra) [inline posthog/hogql/visitor.py:380]: This took a long time to find 🙈

File: posthog/hogql/ast.py [hunk 2]
```diff
@@ -502,22 +537,25 @@ class SelectQuery(Expr):
...
-    offset_value: Optional[RatioExpr]
+    offset_value: Optional[RatioExpr] = None
 
 
-JoinExpr.update_forward_refs(SampleExpr=SampleExpr)
-JoinExpr.update_forward_refs(SelectUnionQuery=SelectUnionQuery)
-JoinExpr.update_forward_refs(SelectQuery=SelectQuery)
+# JoinExpr.update_forward_refs(SampleExpr=SampleExpr)
+# JoinExpr.update_forward_refs(SelectUnionQuery=SelectUnionQuery)
+# JoinExpr.update_forward_refs(SelectQuery=SelectQuery)
```
Comment (thmsobrmlr) [inline posthog/hogql/ast.py:561]: Leftover or Todo?

Comment (mariusandra) [reply posthog/hogql/ast.py:561]: Will remove :).

**Review by Twixes — APPROVED** [review]
Agreed

**Pull request merged**